nemo/collections/tts/data/audio_trimming.py

```diff
@@ -0,0 +1,292 @@
+        sample_rate_ratio = sample_rate / self.vad_sample_rate
+        start_sample_i = sample_rate_ratio * start_sample_i
+        end_sample_i = sample_rate_ratio * end_sample_i
+        return start_sample_i, end_sample_i
+
+    def trim_audio(self, audio: np.array, sample_rate: int, audio_id: str = "") -> Tuple[np.array, int, int]:
+        if sample_rate == self.vad_sample_rate:
+            vad_audio = audio
+        else:
+            # Downsample audio to match sample rate of VAD model
+            vad_audio = librosa.resample(audio, orig_sr=sample_rate, target_sr=self.vad_sample_rate)
+
+        if self.volume_norm:
+            # Normalize volume so we have a fixed scale relative to the reference amplitude
+            vad_audio = normalize_volume(audio=vad_audio, volume_level=1.0)
+
+        speech_frames = self._detect_speech(audio=vad_audio)
+
+        start_i, end_i = get_start_and_end_of_speech(
+            is_speech=speech_frames,
+            frame_threshold=self.frame_threshold,
+            frame_step=self.frame_step,
+            audio_id=audio_id,
+        )
+
+        if sample_rate != self.vad_sample_rate:
+            # Convert sample indices back to input sample rate
+            start_i, end_i = self._scale_sample_indices(start_i, end_i, sample_rate)
+
+        start_i, end_i = pad_sample_indices(
+            start_sample_i=start_i,
+            end_sample_i=end_i,
+            max_sample=audio.shape[0],
+            sample_rate=sample_rate,
+            pad_seconds=self.pad_seconds,
+        )
+
+        trimmed_audio = audio[start_i:end_i]
+
+        return trimmed_audio, start_i, end_i
+
+
+def get_start_and_end_of_speech(
+    is_speech: np.array, frame_threshold: int, frame_step: int, audio_id: str = ""
+) -> Tuple[int, int]:
+    """Finds the start and end of speech for an utterance.
+       Args:
+           is_speech: [num_frames] boolean array with true entries labeling speech frames.
+           frame_threshold: The number of consecutive speech frames required to classify the speech boundaries.
+           frame_step: Audio frame stride used to covert frame boundaries to audio samples.
+           audio_id: String identifier (eg. file name) used for logging.
+
+       Returns integers representing the sample indicies of the start and of speech.
+    """
+    num_frames = is_speech.shape[0]
+
+    # Iterate forwards over the utterance until we find the first frame_threshold consecutive speech frames.
+    start_i = None
+    for i in range(0, num_frames):
+        high_i = min(num_frames, i + frame_threshold)
+        if all(is_speech[i:high_i]):
+            start_i = i
+            break
+
+    # Iterate backwards over the utterance until we find the last frame_threshold consecutive speech frames.
+    end_i = None
+    for i in range(num_frames, 0, -1):
+        low_i = max(0, i - frame_threshold)
+        if all(is_speech[low_i:i]):
+            end_i = i
+            break
```

Comment:
The min/max here might not make sense (I wrote it originally with the check sum() > threshold rather than all()). I think I should stop the iteration when the range becomes less than the threshold size.

Reply:
New version looks fine to me!